### DISCUSSION TITLE
Attachments to ticket

### DISCUSSION THREAD

**Discussion by fabianpena90** [cloudbees/zendesk-java-client#541] [Q&A] [answered]
Hello,

I am working with ZD API but i haven't been able to get it to work when it comes to add attachments to tickets.
This is my code:

```
override fun uploadFileToZendeskAttachmentsApi(fileName: String, file: ByteArray, appUser: AppUser): AttachmentResponse? {
         val url = "$zendeskAttachmentUrl?filename=$fileName"
         val encodedCredentials: String = Base64.getEncoder().encodeToString("$appUser.email:appUser.password".toByteArray())
         val request = httpClient.target(url)
             .request()
             .header("Content-Type", MediaType.TEXT_PLAIN)
             .header("Authorization", "Basic $encodedCredentials")
             .post(Entity.entity(file, MediaType.MULTIPART_FORM_DATA))

         if (request.status == 201) {
            return request.readEntity(AttachmentResponse::class.java)
        } else {
            logger.error("Error executing your request to Zendesk: {${request.status}}, {${request.entity}}")
            throw ZendeskCommunicationException("ERROR_EXECUTING_YOUR_REQUEST_TO_ZENDESK", "Error executing your request to Zendesk")
        }
    }
```

```
AttachmentResponse is my DTO which it looks like this:
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
data class AttachmentResponse(
    val upload: Upload?
)
```

```
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
data class Upload(
    @JsonProperty("id")
    val id: String?,
    @JsonProperty("token")
    val token: String?,
    @JsonProperty("attachment_url")
    val attachmentUrl: String?,
    @JsonProperty("content_url")
    val contentUrl: String?,
    @JsonProperty("expiresAt")
    val expiresAt: Instant?
    )
```

so far i am getting a 400 error request, but i do not know why.

if you know where the error might be please help me :) 

Thanks in advance,
Fabian

**Comment by PierreBtz** [comment, marked as answer]
I'm not sure you are in the correct forum for your question since it's seems related to the Zendesk API, not the java client. I don't know Kotlin so I cannot give you an expert answer, but here are a few notes:
* I would suggest first digging into the content type of your request. IIUC, you are passing `multipart/form-data`, while in the documentation of the Zendesk API (https://developer.zendesk.com/api-reference/ticketing/tickets/ticket-attachments/) they are passing `application/x-www-form-urlencoded`.
* If it doesn't work, then use the curl request of the Zendesk documentation as a reference, activate the logs of whatever http client you are using and compare the request headers with the one of the curl command.
Hope that helps.
